hardware_interface/include/hardware_interface/handle.hpp

```diff
@@ -55,43 +56,64 @@ class Handle
 {
 public:
   [[deprecated("Use InterfaceDescription for initializing the Interface")]]
-  Handle(
-    const std::string & prefix_name, const std::string & interface_name,
-    double * value_ptr = nullptr)
+  Handle(const std::string & prefix_name, const std::string & interface_name, double * value_ptr)
   : prefix_name_(prefix_name),
     interface_name_(interface_name),
     handle_name_(prefix_name_ + "/" + interface_name_),
     value_ptr_(value_ptr)
   {
   }
 
-  explicit Handle(const InterfaceDescription & interface_description)
-  : prefix_name_(interface_description.get_prefix_name()),
-    interface_name_(interface_description.get_interface_name()),
-    handle_name_(interface_description.get_name())
+  explicit Handle(
+    const std::string & prefix_name, const std::string & interface_name,
+    const std::string & data_type = "double", const std::string & initial_value = "")
+  : prefix_name_(prefix_name),
+    interface_name_(interface_name),
+    handle_name_(prefix_name_ + "/" + interface_name_),
+    data_type_(data_type)
   {
-    data_type_ = interface_description.get_data_type();
     // As soon as multiple datatypes are used in HANDLE_DATATYPE
     // we need to initialize according the type passed in interface description
     if (data_type_ == hardware_interface::HandleDataType::DOUBLE)
     {
-      value_ = std::numeric_limits<double>::quiet_NaN();
-      value_ptr_ = std::get_if<double>(&value_);
+      try
+      {
+        value_ = initial_value.empty() ? std::numeric_limits<double>::quiet_NaN()
+                                       : hardware_interface::stod(initial_value);
+        value_ptr_ = std::get_if<double>(&value_);
+      }
+      catch (const std::invalid_argument & err)
+      {
+        throw std::invalid_argument(
+          fmt::format(
+            FMT_COMPILE(
+              "Invalid initial value : '{}' parsed for interface : '{}' with type : '{}'"),
+            initial_value, handle_name_, data_type_.to_string()));
+      }
```

Comment:
there is no test for this assertion, should we add a simple one?

Reply:
Added 👍🏾